catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/writer/CswRecordCollectionMessageBodyWriter.java

```diff
@@ -82,17 +87,27 @@ public void writeTo(CswRecordCollection recordCollection, Class<?> type, Type ge
         final String mimeType = recordCollection.getMimeType();
         LOGGER.debug(
                 "Attempting to transform RecordCollection with mime-type: {} & outputSchema: {}",
-                mimeType, recordCollection.getOutputSchema());
+                mimeType,
+                recordCollection.getOutputSchema());
         QueryResponseTransformer transformer;
         Map<String, Serializable> arguments = new HashMap<String, Serializable>();
-        if (StringUtils.isBlank(recordCollection.getOutputSchema()) && StringUtils
-                .isNotBlank(mimeType) && !XML_MIME_TYPES.contains(mimeType)) {
+        if (StringUtils.isBlank(recordCollection.getOutputSchema()) && StringUtils.isNotBlank(
+                mimeType) && !XML_MIME_TYPES.contains(mimeType)) {
             transformer = transformerManager.getTransformerByMimeType(mimeType);
+        } else if (recordCollection.getOutputSchema()
+                .equals(OCTET_STREAM_OUTPUT_SCHEMA)) {
+            Resource resource = recordCollection.getResource();
```

Comment:
I don’t think it needs to go into the CswQueryResponseTransformer as that class is used for transforming the metadata from the CswRecordCollection into a form of XML represented by the default CSW output schema. All I need to do here is copy the data from the resource object into an OutputStream. No transformer is needed for this.